certbot-apache/certbot_apache/tests/parser_test.py

```diff
@@ -30,6 +43,19 @@ def test_aug_version(self):
         self.parser.aug.match.side_effect = RuntimeError
         self.assertFalse(self.parser.check_aug_version())
 
+    def test_recovery_routine_reload(self):
```

Comment:
I think we should keep this test in `configurator_reverter_test.py` as I think it's testing that the configurators implementation of `recovery_routine` properly asks the parser to reload the config.

Reply:
My approach to this is that it specifically checks that augeas state is reloaded, and because of this I'd like to keep it within the parser tests. When we get to test against the new parser interface, another similar test to `configurator_test.py` should be created, checking that the interface `load()` is called.

Reply:
At the risk of harping on this very minor point too much, I might agree with keeping it in the parser tests if the fact we're calling `load` on Augeas had to do with the code in `parser.py` but I don't think it does. In `configurator.py`, we're reaching into the parser object and calling `parser.aug.load()` directly. I think the only code in `parser.py` that this function is indirectly testing is that the parser has an `aug` attribute which is tested plenty of other places.

I completely agree with you that we want a configurator test asserting that we properly reload the config using the correct interface after a recovery routine, but I think this is that test. We're checking that the implementation of `recovery_routine` in the configurator is properly reloading the parser with its currently existing interfaces. When the parser's interface changes, we'll need to update the code in `configurator.py` to use those interfaces as well as the tests for that configurator code. We'll also probably want a parser test in the future that tests that the common `load()` interface properly calls `load()` on Augeas, but I don't think this test needs to be coupled to the configurator code.

For those reasons, I personally think it makes sense to just have a single test for reloading during `recovery_routine` and for this test to exist in `configurator_reverter_test.py`.

If you disagree with me, I'd kind of like to know why, but I don't think it's worth blocking this PR on this minor point. If you still prefer to keep the test here after my additional explanation of how I'm thinking about this, let me know and I'll merge this PR as is.

Reply:
Yeah, I agree. I was too focused on keeping the `configurator_reverter_test.py` parser implementation agnostic (for the changes to come). It is the logical place for this test, and we'll have to refactor tests a lot when we're finally removing Augeas anyway. This test is now moved to the reverter test.